api/swagger.yaml

```diff
@@ -157,6 +157,293 @@ tags:
+            MaxFailureRatio: 0.15
+          RollbackConfig:
+            Parallelism: 1
+            Delay: 1000000000
+            FailureAction: "pause"
+            Monitor: 15000000000
+            MaxFailureRatio: 0.15
+          EndpointSpec:
+            Mode: "vip"
+  ServiceCreateBody:
+    allOf:
+      - $ref: "#/definitions/ServiceSpec"
+      - type: "object"
+        example:
+          Name: "web"
+          TaskTemplate:
+            ContainerSpec:
+              Image: "nginx:alpine"
+              Mounts:
+                -
+                  ReadOnly: true
+                  Source: "web-data"
+                  Target: "/usr/share/nginx/html"
+                  Type: "volume"
+                  VolumeOptions:
+                    DriverConfig: {}
+                    Labels:
+                      com.example.something: "something-value"
+              Hosts: ["10.10.10.10 host1", "ABCD:EF01:2345:6789:ABCD:EF01:2345:6789 host2"]
+              User: "33"
+              DNSConfig:
+                Nameservers: ["8.8.8.8"]
+                Search: ["example.org"]
+                Options: ["timeout:3"]
+              Secrets:
+                -
+                  File:
+                    Name: "www.example.org.key"
+                    UID: "33"
+                    GID: "33"
+                    Mode: 384
+                  SecretID: "fpjqlhnwb19zds35k8wn80lq9"
+                  SecretName: "example_org_domain_key"
+            LogDriver:
+              Name: "json-file"
+              Options:
+                max-file: "3"
+                max-size: "10M"
+            Placement: {}
+            Resources:
+              Limits:
+                MemoryBytes: 104857600
+              Reservations: {}
+            RestartPolicy:
+              Condition: "on-failure"
+              Delay: 10000000000
+              MaxAttempts: 10
+          Mode:
+            Replicated:
+              Replicas: 4
+          UpdateConfig:
+            Parallelism: 2
+            Delay: 1000000000
+            FailureAction: "pause"
+            Monitor: 15000000000
+            MaxFailureRatio: 0.15
+          RollbackConfig:
+            Parallelism: 1
+            Delay: 1000000000
+            FailureAction: "pause"
+            Monitor: 15000000000
+            MaxFailureRatio: 0.15
+          EndpointSpec:
+            Ports:
+              -
+                Protocol: "tcp"
+                PublishedPort: 8080
+                TargetPort: 80
+          Labels:
+            foo: "bar"
+  SecretCreateBody:
+    allOf:
+      - $ref: "#/definitions/SecretSpec"
+      - type: "object"
+        example:
+          Name: "app-key.crt"
+          Labels:
+            foo: "bar"
+          Data: "VEhJUyBJUyBOT1QgQSBSRUFMIENFUlRJRklDQVRFCg=="
+          Driver:
+            Name: "secret-bucket"
+            Options:
+              OptionA: "value for driver option A"
+              OptionB: "value for driver option B"
+  ConfigCreateBody:
```

Comment:
Looking at these definitions, I don't think there's a need to create a new definition. The only `allOf` used here looks to be adding example values (it doesn't add new properties to the existing definition).

Instead (in this case for `ConfigSpec`), just adding the example values to the `ConfigSpec` type should probably work:

```yaml
  ConfigSpec:
    type: "object"
    properties:
      Name:
        description: "User-defined name of the config."
        type: "string"
        example: "server.conf"
      Labels:
        description: "User-defined key/value metadata."
        type: "object"
        additionalProperties:
          type: "string"
        example:
          foo: "bar"
      Data:
        description: |
          Base64-url-safe-encoded ([RFC 4648](https://tools.ietf.org/html/rfc4648#section-3.2))
          config data.
        type: "string"
        example: "VEhJUyBJUyBOT1QgQSBSRUFMIENFUlRJRklDQVRFCg=="
```

Or, diff:

```patch
diff --git a/api/swagger.yaml b/api/swagger.yaml
index 2e4615f77..08a5286a5 100644
--- a/api/swagger.yaml
+++ b/api/swagger.yaml
@@ -3354,16 +3354,20 @@ definitions:
       Name:
         description: "User-defined name of the config."
         type: "string"
+        example: "server.conf"
       Labels:
         description: "User-defined key/value metadata."
         type: "object"
         additionalProperties:
           type: "string"
+        example:
+          foo: "bar"
       Data:
         description: |
           Base64-url-safe-encoded ([RFC 4648](https://tools.ietf.org/html/rfc4648#section-3.2))
           config data.
         type: "string"
+        example: "VEhJUyBJUyBOT1QgQSBSRUFMIENFUlRJRklDQVRFCg=="

   Config:
     type: "object"
@@ -9720,14 +9724,7 @@ paths:
         - name: "body"
           in: "body"
           schema:
-            allOf:
-              - $ref: "#/definitions/ConfigSpec"
-              - type: "object"
-                example:
-                  Name: "server.conf"
-                  Labels:
-                    foo: "bar"
-                  Data: "VEhJUyBJUyBOT1QgQSBSRUFMIENFUlRJRklDQVRFCg=="
+            $ref: "#/definitions/ConfigSpec"
       tags: ["Config"]
   /configs/{id}:
     get:
```

A quick look at the other cases looks like a similar approach could be used there as well